src/peft/peft_model.py

```diff
@@ -1185,6 +1200,19 @@ def load_adapter(
             ignore_mismatched_sizes=ignore_mismatched_sizes,
             low_cpu_mem_usage=low_cpu_mem_usage,
         )
+        missing_keys, unexpected_keys = load_result.missing_keys, load_result.unexpected_keys
+        tuner = self.peft_config[adapter_name].peft_type
+
+        tuner_prefix = PEFT_TYPE_TO_PREFIX_MAPPING.get(tuner, "")
+        adapter_missing_keys = []
+
+        # Filter missing keys specific to the current adapter and tuner prefix.
+        for key in missing_keys:
+            if tuner_prefix in key and adapter_name in key:
+                adapter_missing_keys.append(key)
+
+        load_result = _IncompatibleKeys(missing_keys=adapter_missing_keys, unexpected_keys=unexpected_keys)
```

Comment:
I guess you probably first tried to override `load_result.missing_keys` and as that doesn't work, you decided to create a new instance of `_IncompatibleKeys`. However, this is not necessary. You can actually mutate `load_result.missing_keys`, i.e. you can call `clear`, `append`, and `extend` on it.